### PR TITLE
Add section on API/ABI evolution.

### PR DESCRIPTION
**Pull request by rgommers**

### CODE REVIEW

File: scipy-1.0/paper.tex [hunk 1]
```diff
@@ -225,10 +225,26 @@ \subsection*{Language choices}
...
+\begin{itemize}
+    \item The algorithm is of relevance to multiple fields of science.
+    \item The implementation is based on a peer-reviewed article, and that article's importance is demonstrated (e.g. by having a significant number of citations).
+\end{itemize}
+
+Our policy for evolving the API over time is that new functionality can be added,
+while removing or changing existing functionality can only be done if the benefits
+of that exceeds the (often significant) costs to users, \textit{and} only
+after giving clear deprecation warnings to those users for at least one year.
+In general, we encourage changes that improveclarity in the API of the
```
Comment (larsoner) [inline scipy-1.0/paper.tex]: improveclarity -> improve clarity

File: scipy-1.0/paper.tex [hunk 2]
```diff
@@ -225,10 +225,26 @@ \subsection*{Language choices}
...
-In general, we encourage changes that improve clarity in the API of the library,
-but strongly discourage breaking backwards compatibility, given our position near
-the base of the scientific Python computing stack.
-
+The application programming interface (API) for SciPy consists of
+approximately 2600 functions and classes. Adding new functionality has to
+meet the following criteria:
+\begin{itemize}
+    \item The algorithm is of relevance to multiple fields of science.
+    \item The implementation is based on a peer-reviewed article, and that article's importance is demonstrated (e.g. by having a significant number of citations).
```
Comment (ev-br) [inline scipy-1.0/paper.tex]: Minor: maybe generalize this a little bit:  it sounds as if we would not consider an algorithm which is classic enough be in textbooks.

File: scipy-1.0/paper.tex [hunk 3]
```diff
@@ -225,10 +225,26 @@ \subsection*{Language choices}
...
+\end{itemize}
+
+Our policy for evolving the API over time is that new functionality can be added,
+while removing or changing existing functionality can only be done if the benefits
+of that exceeds the (often significant) costs to users, \textit{and} only
+after giving clear deprecation warnings to those users for at least one year.
+In general, we encourage changes that improve clarity in the API of the
+library, but strongly discourage breaking backwards compatibility, given our
```
Comment (ev-br) [inline scipy-1.0/paper.tex]: Minor: maybe spell out what backwards compatibility is: the term is not necessarily familiar to non-CS readers.

File: scipy-1.0/paper.tex (hunk 2 above)
Comment (rgommers) [reply scipy-1.0/paper.tex]: good point, done

File: scipy-1.0/paper.tex (hunk 3 above)
Comment (rgommers) [reply scipy-1.0/paper.tex]: added as a footnote

**Pull request merged**